pennylane_lightning_kokkos/src/simulator/StateVectorKokkos.cpp

```diff
@@ -17,3 +17,19 @@
 // explicit instantiation
 template class Pennylane::StateVectorKokkos<float>;
 template class Pennylane::StateVectorKokkos<double>;
+
+std::string repr_InitArguments(const Kokkos::InitArguments &a) {
+    std::string str;
+    str = "<example.InitArguments with";
+    str += "\n num_threads = " + std::to_string(a.num_threads);
+    str += "\n num_numa = " + std::to_string(a.num_numa);
+    str += "\n device_id = " + std::to_string(a.device_id);
+    str += "\n ndevices = " + std::to_string(a.ndevices);
+    str += "\n skip_device = " + std::to_string(a.skip_device);
+    str += "\n disable_warnings = " + std::to_string(a.disable_warnings) += ">";
```

Comment:
The whole string could be defined in one go. We could also call the `reserve` method to preallocate space for the string. We could also just allocate a C char array and `sprintf`.

Reply:
Yep, reserve here would definitely help since we have a good general idea on the number of characters up-front.

Another method would be to make use of an `std::ostream` type, or equivalent. With this, we can stream the data into the object, and then generate the string entirely at the end. While there can be overheads with this approach, it is generally the preferred way. See https://github.com/PennyLaneAI/pennylane-lightning-kokkos/blob/279ba340d34607fc4f83b0ae81c3e2e53345cb86/pennylane_lightning_kokkos/src/util/Util.hpp#L267 for an example on how we define this for a vector type via the suggested method from @AmintorDusko  earlier.